sov-modules/sov-state/src/value.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{storage::StorageKey, Prefix, StateMap, Storage};
+use sovereign_sdk::serial::{Decode, Encode};
+use std::marker::PhantomData;
+
+/// SingletonKey is very similar to the unit type `()` i.e. it has only one value.
+/// We provide a custom efficient Encode implementation for SingletonKey while Encode for `()`
+/// is likely already implemented by an external library (like borsh), which is outside of our control.
+#[derive(Debug)]
+pub struct SingletonKey;
+
+impl Encode for SingletonKey {
+    fn encode(&self, _: &mut impl std::io::Write) {
+        // Do nothing.
+    }
+}
```

Comment:
Note, non-blocking: Since we manually implement encode, any types containing SingletonKey won't be able to use the derive macros

Reply:
I made the type private to the module so it won't leak (before I needed it in tests in another crate, but I found a walk around).  Here we won't mix it with other types.